tests/examples/components/shared/test_aml_smpc.py

```diff
@@ -0,0 +1,115 @@
+import os
+import sys
+import unittest
+
+from cryptography.hazmat.primitives import serialization as crypto_serialization
+
+sys.path.append(
+    os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "..", "..", ".."))
+)
+from examples.components.shared.aml_smpc import AMLSMPC
+
+
+class TestAMLSMPC(unittest.TestCase):
+    def test_aml_smpc_basic(self):
+        TEST_MSG = lambda recv: f"Message to {recv}"
+
+        smpc1 = AMLSMPC()
+        smpc2 = AMLSMPC()
+
+        smpc1.add_remote_public_key(2, smpc2.get_public_key())
+        smpc2.add_remote_public_key(1, smpc1.get_public_key())
+
+        msg_2 = smpc1.encrypt(str.encode(TEST_MSG(2)), 2)
+        msg_1 = smpc2.encrypt(str.encode(TEST_MSG(1)), 1)
+
+        assert smpc1.decrypt(msg_1) == str.encode(TEST_MSG(1))
+        assert smpc2.decrypt(msg_2) == str.encode(TEST_MSG(2))
+
+    def test_aml_smpc_encodings_formats(self):
```

Comment:
Same comment as in test_aml_comm.py: is it possible to parameterize to avoid code duplication? (and explicitly log what kind of test is being performed, and the reuslts)

Reply:
I increased verbosity of an output - what do you mean by code duplication?

Reply:
Right now, the code in your `test_aml_smpc_basic()` function is exactly the same as the code in the subtest of your `test_aml_smpc_encodings_formats()` function (minus the encoding and format parameters of course). The code for `test_aml_smpc_empty()` is the same too (minus the message generation of course). Why not do it all in one single function using subtests?

I wouldn't worry about the last 2 test functions on overwrite though. They could probably be combined into a single function, but I guess that would require some ugly if statements and make the code harder to read